heudiconv/dicoms.py

```diff
@@ -570,7 +582,7 @@ def _assign_dicom_time(ti: tarfile.TarInfo) -> tarfile.TarInfo:
 def embed_dicom_and_nifti_metadata(
     dcmfiles: List[str],
     niftifile: str,
-    infofile: str | Path,
+    infofile: Union[str, Path],
```

Comment:
how come we had tests passed on python 3.7? we do have `from __future__ import annotations` on top

Reply:
* Read the comment above `embed_dicom_and_nifti_metadata()` about why `from __future__ ...` in our source file is irrelevant.

* I'm not sure why the tests passed, but I suspect it may be related to #667.